lstchain/high_level/tests/test_interpolate.py

```diff
@@ -0,0 +1,359 @@
+    data_pars2 = {"ZEN_PNT": 58 * u.deg, "B_DELTA": 70 * u.deg, "AZ_PNT": 200 * u.deg}
+
+    new_irfs = check_in_delaunay_triangle(irfs, data_pars)
+    new_irfs2 = check_in_delaunay_triangle(irfs, data_pars2)
+    new_irfs3 = check_in_delaunay_triangle(
+        irfs, data_pars, use_nearest_irf_node=True
+    )
+    new_irfs4 = check_in_delaunay_triangle([irfs[0]], data_pars)
+
+    t3 = Table.read(new_irfs3[0], hdu=1).meta
+
+    assert len(new_irfs) == 3
+    assert len(new_irfs2) == 1
+    assert t3["ZEN_PNT"] == 20
+    assert len(new_irfs4) == 1
+
+
+def test_get_nearest_az_node():
+    from scipy.spatial import distance
+
+    from lstchain.high_level.interpolate import get_nearest_az_node, interp_params
+
+    # 2 coincident nodes in IRF interpolation grid (cos zenith, sin delta)
+    data_0 = {
+        "ZEN_PNT": 10 * u.deg,
+        "B_DELTA": 50.3607 * u.deg,
+        "AZ_PNT": 102.199 * u.deg,
+    }
+    data_1 = {
+        "ZEN_PNT": 10 * u.deg,
+        "B_DELTA": 50.3607 * u.deg,
+        "AZ_PNT": 248.117 * u.deg,
+    }
+    # Target node, with same b_delta (sin delta) value as the above nodes
+    # AZ_PNT and ZEN_PNT are random close values, not physical.
+    data_target = {
+        "ZEN_PNT": 12 * u.deg,
+        "B_DELTA": 50.3607 * u.deg,
+        "AZ_PNT": 150 * u.deg,
+    }
+    params_list = ["ZEN_PNT", "B_DELTA"]
+    params_list_full = ["ZEN_PNT", "B_DELTA", "AZ_PNT"]
+    az_idx = list(data_target.keys()).index("AZ_PNT")
+
+    # Use interp_params to have appropriate lists for the test
+    # Without AZ_PNT, for IRF interpolation space
+    check_params_0 = interp_params(params_list, data_0)
+    check_params_1 = interp_params(params_list, data_1)
+    check_params_target = interp_params(params_list, data_target)
+    check_params = np.array([check_params_0, check_params_1])
+
+    # With AZ_PNT
+    check_params_0_full = interp_params(params_list_full, data_0)
+    check_params_1_full = interp_params(params_list_full, data_1)
+    check_params_target_full = interp_params(params_list_full, data_target)
+    check_params_full = np.array([check_params_0_full, check_params_1_full])
+
+    # Get the distance between the target and given nodes, in IRF interpolation grid
+    index = distance.cdist([check_params_target], check_params).argmin()
+
+    # Fetch the index of node, closest to the target
+    index2 = get_nearest_az_node(
+        check_params, index, check_params_full, check_params_target_full, az_idx
+    )
+
+    # Closest node
+    mc_closest = check_params_full[index2]
+
+    # Values to compare
+    mc_closest_az = round(mc_closest[-1], 3)
+    az_check = round(102.199 * np.pi / 180, 3)
+
+    assert mc_closest_az == az_check
+
+
+def test_interpolate_gh_cuts():
+    from lstchain.high_level.interpolate import interpolate_gh_cuts
+
+    # Similar function as interpolate_th_cuts, hence no need for extra test
+    # linear test case
```

Comment:
Not sure what is the use of duplicated code like this. You created 2 different functions which do exactly the same thing and only rename (and order?) the call to an external method.

Still if you do it like this you should cover both with test and check they always give the same output. When maintaining the code it will allow to avoid unintended behaviour by modifying one and not the other.

Reply:
The `interpolate_rad_max` will be imported from pyirf, and when we update the pyirf dependency, we will not have an extra test for it. The current function is just a placeholder so that users can use the PR, without updating pyirf.

Adding an extra test is not an issue, but it just felt like a redundant effort...

I will fix the typo in the name of the function `interpolate_rad_max` here.

Reply:
Ok no problem if it is temporary.